### PR TITLE
OSD-9942 Produce local metrics

### PR DESCRIPTION
**Pull request by mrbarge**
### What type of PR is this?
Feature

### What this PR does / why we need it?
Implements the two `ocm-agent-operator` metrics identified by [OSD-9566](https://issues.redhat.com/browse/OSD-9566):
- `ocmagentoperator_ocm_agent_pull_secret_invalid`
- `ocmagentoperator_ocm_agent_resource_absent`

### Which Jira/Github issue(s) this PR fixes?
[OSD-9942](https://issues.redhat.com/browse/OSD-9942)

### Special notes for your reviewer:
Test cases have been written for `ocmagentoperator_ocm_agent_pull_secret_invalid` but the nature of `ocm_agent_resource_absent` and where it is set (the reconcile function) doesn't currently lend itself well to testing.

### Pre-checks (if applicable):
- [X] Tested latest changes against a cluster
- [X] Ran `make generate` command locally to validate code changes
- [X] Included documentation changes with PR

### CODE REVIEW

File: pkg/localmetrics/localmetrics.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		Subsystem: metricsTag,
+		Name:      "ocm_agent_pull_secret_invalid",
+		Help:      "Failed to obtain a valid pull secret",
+	}, []string{nameLabel})
+
+	MetricOcmAgentResourceAbsent = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Subsystem: metricsTag,
+		Name:      "ocm_agent_resource_absent",
+		Help:      "No OCM Agent resource found",
+	}, []string{})
```
Comment (bmeng) [inline pkg/localmetrics/localmetrics.go:23]: Maybe we can have a label here to reflect the resource type?
But I doubt if the condition here would happen anyway.

Comment (mrbarge) [reply pkg/localmetrics/localmetrics.go:23]: I figured the resource type is kind of implied by the metric name of `ocm_agent_resource`. 
But yeah.. I am in agreement that this metric should be almost always 0.

File: pkg/ocmagenthandler/ocmagenthandler_secret.go [hunk 2]
```diff
@@ -36,8 +37,10 @@ func (o *ocmAgentHandler) ensureAccessTokenSecret(ocmAgent ocmagentv1alpha1.OcmA
 
 	clusterPullSecret, err := o.fetchAccessTokenPullSecret()
 	if err != nil {
+		localmetrics.UpdateMetricPullSecretInvalid(ocmAgent.Name)
```
Comment (bmeng) [inline pkg/ocmagenthandler/ocmagenthandler_secret.go:40]: seems the metric here indicates only the format of the pull secret, not sure if we can valid the contains.
Out of scope but right my thoughts down here.
Maybe move this to the ocm-agent, when the ocm request return code is 40x in the webhookreceiver.

Comment (mrbarge) [reply pkg/ocmagenthandler/ocmagenthandler_secret.go:40]: I'd have to review the PR I made to the OCM SDK, but I think it (the OCM SDK) might will raise an error if a malformed token is supplied. In which case yeah we can have ocm-agent produce more detailed metrics there.

**Pull request merged**